### PR TITLE
Add image classification fine-tuning notebook in TensorFlow 

### PR DESCRIPTION
**Pull request by sayakpaul**
@amyeroberts

### CODE REVIEW

**Comment by amyeroberts** [comment]
Thanks for your work on this contribution @sayakpaul ! 

The diff is too large to render on the github PR and as such I can't comment directly on the code. So will comment here with references to make it as clear / easy as possible. Let me know if there's an alternative way you'd like this to be done.

Comments:
* This PR has your forked repo's main as a target. Could you change this to `huggingface:main`? 

* Overall, nicely laid out and clear examples and code :) 

* This PR adds two notebooks `examples/image-classification-tf.ipynb` and `examples/image_classification_tf.ipynb`. `examples/image_classification_tf.ipynb` appears to be the complete one. I suggest: 
	* removing `examples/image-classification-tf.ipynb`
	* renaming `examples/image_classification_tf.ipynb` ->  `examples/image-classification-tf.ipynb` to be consistent with the rest of the repo

* In the `Data Augmentation` section, I would rephrase `This notebook leverages the ops offered by the tf.image module for data augmentation. We will soon be adding data augmentation support with libraries like albumentations.` to something like `This notebook leverages the ops offered by the tf.image module for data augmentation. We will soon be adding examples of data augmentation with libraries like albumentations.` as we can already use albumentations with tf.data.Dataset objects :). 

* In the `Preparing the model` section, I would include the comments in the `image_classification.ipynb`:
    * explaining `ignore_mismatched_sizes` argument when finetuning "NOTE: in case you're planning to fine-tune an already fine-tuned checkpoint, ..."
    * explaining the warnings about loading the model: "The warning is telling us we are throwing away some weights... " 

* In most of our notebooks, we have the same datatset and model used across different framework examples e.g. [language_modeling.ipynb](https://github.com/sayakpaul/notebooks/tree/feat/img-cls-tf-nb/examples#:~:text=language_modeling.ipynb) and [language_modeling-tf.ipynb](https://github.com/sayakpaul/notebooks/tree/feat/img-cls-tf-nb/examples#:~:text=language_modeling%2Dtf.ipynb). Would it be possible to do this (use EuroSAT from the ImageFolder) with the augmentations you've outlined? [Using the Hugging Face `Dataset` class, we can convert to a tf.data.Dataset class](https://huggingface.co/docs/datasets/v2.2.1/en/package_reference/main_classes#datasets.Dataset.to_tf_dataset).  

* It would be great to add an `Inference` section where we show `.predict()` calls on the model. Maybe something to add in a second PR?

* I would add a wee comment somewhere mentioning that this notebook has been written using your notebook as a reference `https://colab.research.google.com/github/sayakpaul/TF-2.0-Hacks/blob/master/data2vec_vision_image_classification.ipynb` (give yourself credit!)

**Comment by sayakpaul** [comment]
@amyeroberts thanks for your comments. 

> The diff is too large to render on the github PR and as such I can't comment directly on the code.

ReviewNB should make this easier. Refer to this comment: https://github.com/sayakpaul/notebooks/pull/1#issuecomment-1139631689

> This PR has your forked repo's main as a target. Could you change this to huggingface:main?

Yes, I should have done it. Will close this PR after one more review from your end. Can do that now too. 

> explaining ignore_mismatched_sizes argument when finetuning "NOTE: in case you're planning to fine-tune an already fine-tuned checkpoint, ..."

I couldn't find this in the TF notebook. Were you referring to the TF notebook? 

> In most of our notebooks, we have the same datatset and model used across different framework examples e.g. [language_modeling.ipynb](https://github.com/sayakpaul/notebooks/tree/feat/img-cls-tf-nb/examples#:~:text=language_modeling.ipynb) and [language_modeling-tf.ipynb](https://github.com/sayakpaul/notebooks/tree/feat/img-cls-tf-nb/examples#:~:text=language_modeling%2Dtf.ipynb). Would it be possible to do this (use EuroSAT from the ImageFolder) with the augmentations you've outlined? [Using the Hugging Face Dataset class, we can convert to a tf.data.Dataset class](https://huggingface.co/docs/datasets/v2.2.1/en/package_reference/main_classes#datasets.Dataset.to_tf_dataset).

I thought about it. I think our users (especially the ones who use TensorFlow) would benefit from an example that marries `transformers` for modeling and `tf.image` ops (and [Keras preprocessing layers](https://keras.io/api/layers/preprocessing_layers)) for data preprocessing. We can also work on another notebook that shows the use of `datasets` with the model we're using in this PR. WDYT? 

Apart from that, I think I have reflected on all the other comments. PTAL.

**Comment by amyeroberts** [comment]
> ReviewNB should make this easier. Refer to this comment: https://github.com/sayakpaul/notebooks/pull/1#issuecomment-1139631689

Awesome - thanks. I don't have any other comments on the diff atm. Let's open the one merging into `huggingface:main`

> Yes, I should have done it. Will close this PR after one more review from your end. Can do that now too.

Great - please add a comment on the new PR linking to this one for anyone that wants to see the history.

> > explaining ignore_mismatched_sizes argument when finetuning "NOTE: in case you're planning to fine-tune an already fine-tuned checkpoint, ..."

>i couldn't find this in the TF notebook. Were you referring to the TF notebook?

These comments can be found in the [pytorch image classification notebook](https://github.com/sayakpaul/notebooks/blob/feat/img-cls-tf-nb/examples/image_classification.ipynb)

> I thought about it. I think our users (especially the ones who use TensorFlow) would benefit from an example that marries transformers for modeling and tf.image ops (and [Keras preprocessing layers](https://keras.io/api/layers/preprocessing_layers)) for data preprocessing. We can also work on another notebook that shows the use of datasets with the model we're using in this PR. WDYT?

Generally, I think it's better to have consistent examples between the notebooks. It demonstrates that the same tasks can be done with different frameworks and highlights any implementation differences needed to achieve this. 

That being said, it's useful to demonstrate using a `tf.data.Dataset` object as that is what many people use. I've done a bit of digging and there's some more work needed in our data_collators to successfully call `to_tf_dataset` so lets keep this example for now :) 

As a side note - I completely agree having `tf.image` as part of the data processing steps is useful for our users. I don't think using a hugging face dataset and tf.image processing are mutually exclusive :) we can use it in a processing function that's applied using `set_transform`

**Comment by sayakpaul** [comment]
@amyeroberts https://github.com/huggingface/notebooks/pull/197
